### PR TITLE
Clarify error messages for functions

### PR DESCRIPTION
**Pull request by laurenceisla**
Moves the explanations about "single unnamed parameters" to the details instead of the message part of the error. It also removes any mention to resetting the schema cache from other error hints.

### CODE REVIEW

File: src/PostgREST/Error.hs [hunk 1]
```diff
@@ -181,21 +181,22 @@ instance JSON.ToJSON ApiRequestError where
...
     let func = schema <> "." <> procName
         prms = "(" <> T.intercalate ", " argumentKeys <> ")"
-        fmtParams =  if null argumentKeys then " function without parameters" else prms <> " function"
+        fmtParams =  if null argumentKeys then " without parameters" else prms
+        onlySingleParams = hasPreferSingleObject || (isInvPost && contentType `elem` [MTTextPlain, MTTextXML, MTOctetStream])
     in JSON.object [
     "code"    .= SchemaCacheErrorCode02,
-    "message" .= ("Could not find the " <> func <>
+    "message" .= ("Could not find the function " <> func <> (if onlySingleParams then "" else fmtParams) <> " in the schema cache"),
+    "details" .= ("Looked for the function " <> func <>
```
Comment (steve-chavez) [inline src/PostgREST/Error.hs:189]:
```suggestion
    "details" .= ("Searched for the function " <> func <>
```

File: src/PostgREST/Error.hs [hunk 2]
```diff
@@ -171,7 +171,7 @@ instance JSON.ToJSON ApiRequestError where
     "code"    .= SchemaCacheErrorCode00,
     "message" .= ("Could not find a relationship between '" <> parent <> "' and '" <> child <> "' in the schema cache" :: Text),
     "details" .= JSON.Null,
-    "hint"    .= ("Verify that '" <> parent <> "' and '" <> child <> "' exist in the schema '" <> schema <> "' and that there is a foreign key relationship between them. If a new relationship was created, try reloading the schema cache." :: Text)]
+    "hint"    .= ("Verify that '" <> parent <> "' and '" <> child <> "' exist in the schema '" <> schema <> "' and that there is a foreign key relationship between them." :: Text)]
```
Comment (steve-chavez) [inline src/PostgREST/Error.hs:174]: This seems unrelated, how about leaving this change for another PR?(dedicated to closing https://github.com/PostgREST/postgrest/issues/2569#issuecomment-1334540506)

Comment (steve-chavez) [reply src/PostgREST/Error.hs:174]: ^ easier to review if the change is focused

Comment (laurenceisla) [reply src/PostgREST/Error.hs]: Agree, done.

File: test/spec/Feature/Query/RpcSpec.hs [hunk 3]
```diff
@@ -1274,9 +1274,9 @@ spec actualPgVersion =
             `shouldRespondWith`
               [json|{
                 "hint": "Perhaps you meant to call the function test.unnamed_text_param",
-                "message": "Could not find the test.unnamed_int_param(x, y) function or the test.unnamed_int_param function with a single unnamed json or jsonb parameter in the schema cache",
+                "message": "Could not find the function test.unnamed_int_param(x, y) in the schema cache",
                 "code":"PGRST202",
-                "details":null
+                "details":"Searched for the function test.unnamed_int_param(x, y) or with a single unnamed json/jsonb parameter, but no matches were found in the schema cache."
```
Comment (steve-chavez) [inline test/spec/Feature/Query/RpcSpec.hs:1279]: Hm, this is not completely clear.

Maybe we should decompress the "(x,y)" part, this would also clarify what we mean on the message. Otherwise it seems redundant.

Comment (steve-chavez) [reply test/spec/Feature/Query/RpcSpec.hs:1279]:
```suggestion
                "details":"Searched for the function test.unnamed_int_param with x, y parameters or with a single unnamed json/jsonb parameter, but no matches were found in the schema cache."
```

Comment (laurenceisla) [reply test/spec/Feature/Query/RpcSpec.hs:1279]: Maybe something like this?

```suggestion
                "details":"Searched for the function test.unnamed_int_param with parameters x, y or with a single unnamed json/jsonb parameter, but no matches were found in the schema cache."
```

Comment (steve-chavez) [reply test/spec/Feature/Query/RpcSpec.hs:1279]: Yes, that would be better.

File: src/PostgREST/Error.hs [hunk 4]
```diff
@@ -180,22 +180,25 @@ instance JSON.ToJSON ApiRequestError where
     "hint"    .= ("Try changing '" <> child <> "' to one of the following: " <> relHint rels <> ". Find the desired relationship in the 'details' key." :: Text)]
   toJSON (NoRpc schema procName argumentKeys hasPreferSingleObject contentType isInvPost allProcs overloadedProcs)  =
     let func = schema <> "." <> procName
-        prms = "(" <> T.intercalate ", " argumentKeys <> ")"
-        fmtParams =  if null argumentKeys then " function without parameters" else prms <> " function"
+        prms = T.intercalate ", " argumentKeys
+        prmsMsg = "(" <> prms <> ")"
+        prmsDet = " with parameter(s) " <> prms
```
Comment (steve-chavez) [inline src/PostgREST/Error.hs:185]: That parenthesis is confusing, the message already uses it for compressing the parameters.
```suggestion
        prmsDet = " with parameters " <> prms
```

Comment (laurenceisla) [reply src/PostgREST/Error.hs]: True, I made it more accurate now.

**Review by steve-chavez — APPROVED** [review]
